test/topology/util.py

```diff
@@ -13,7 +13,7 @@
 import operator
 import time
 import re
-from contextlib import suppress
+from contextlib import asynccontextmanager, contextmanager, suppress
```

Comment:
How can we add the test before the tested feature is implemented?

Reply:
The test has been added with `xfail` (expected to fail) first - the idea being that the test is expected to fail first without the feature, and then succeed once the feature is implemented.

Reply:
But it is also worth to say that the test in fact evolved during the implementation (as I didn't know how it will actually behave upfront indeed) so the final test version has been updated then accordingly (to still make sure it fails without the feature).

Reply:
ok, that is fine